virttest/qemu_devices/qcontainer.py

```diff
@@ -4031,9 +4031,41 @@ def hostdev_define_by_params(self, name, params, bus=None):
                 "failover_pair_id": params.get("vm_hostdev_failover_pair_id"),
             }
         )
+
+        devs = []
+
+        iommufd_id = params.objects("vm_hostdev_iommufd")
+        # To support single/multiple iommufd for single/multiple hostdev
+        for fd_id in iommufd_id:
+            if not self.get_by_qid(fd_id):
+                iommufd = self.iommufd_object_define_by_params(fd_id)
```

Comment:
The iommufd maybe create fail, right? for example IOMMUFD driver not loaded and no /dev/iommu.

It needs a check for object before adding iommufd for device.

Reply:
@xhao22 Thanks for reminder, updated in iommufd_object_define_by_param, raise error if /dev/iommu doesn't exist.